### ISSUE TITLE
slash in attribute name causes write error error

### ISSUE THREAD

**Issue by bkmartinjr** [TileDB-Inc/TileDB-Py#294]
Test program below demonstrates failed attempt to persist an array when attribute name contains a '/'    It appears that the attribute name is literally used as the `.tdb` file/object name.

Test case:
```
# bad attr name bug
import numpy as np
import tiledb

col_name = "fo/o"

attrs = [tiledb.Attr(name=col_name, dtype=np.int)]
domain = tiledb.Domain(tiledb.Dim(domain=(0, 99), tile=100, dtype=np.uint32))
schema = tiledb.ArraySchema(domain=domain, sparse=False, attrs=attrs, cell_order="row-major", tile_order="row-major")
tiledb.DenseArray.create("foo", schema)

with tiledb.DenseArray("foo", mode="w") as A:
    value = {}
    value[col_name] = np.zeros((100,), dtype=np.int)
    A[:] = value

tiledb.consolidate("foo")

print("done")
```

Result run on a posix file system where '/' is a reserved character:
```
# python bkm/badattrname.py 
Traceback (most recent call last):
  File "bkm/badattrname.py", line 15, in <module>
    A[:] = value
  File "tiledb/libtiledb.pyx", line 4184, in tiledb.libtiledb.DenseArrayImpl.__setitem__
  File "tiledb/libtiledb.pyx", line 349, in tiledb.libtiledb._write_array
  File "tiledb/libtiledb.pyx", line 413, in tiledb.libtiledb._raise_ctx_err
  File "tiledb/libtiledb.pyx", line 398, in tiledb.libtiledb._raise_tiledb_error
tiledb.libtiledb.TileDBError: [TileDB::IO] Error: Cannot open file '/mnt/c/Users/bruce/projects/cellxgene/foo/__1585662267752_1585662267752_2e82b38a5b9a493088d92bc55cb8f91f/fo/o.tdb'; No such file or directory
```

---

Will be fixed in TileDB 2.3 by https://github.com/TileDB-Inc/TileDB/pull/2107

**Comment by ihnorton** [comment]
Hi @bkmartinjr, we are aiming to add encoding in the next release to make this (and similar situations) safe.

**Comment by nguyenv** [comment]
```
(clean-tiledb) vivian@mangonada:~$ cat bad_attr_name_bug.py
import numpy as np
import tiledb

col_name = "fo/o"

attrs = [tiledb.Attr(name=col_name, dtype=np.int)]
domain = tiledb.Domain(tiledb.Dim(domain=(0, 99), tile=100, dtype=np.uint32))
schema = tiledb.ArraySchema(domain=domain, sparse=False, attrs=attrs, cell_order="row-major", tile_order="row-major")
tiledb.DenseArray.create("foo", schema)

with tiledb.DenseArray("foo", mode="w") as A:
    value = {}
    value[col_name] = np.zeros((100,), dtype=np.int)
    A[:] = value

tiledb.consolidate("foo")

print("done")
(clean-tiledb) vivian@mangonada:~$ python bad_attr_name_bug.py
done
```

This bug has been fixed in TileDB-Py 0.9 / TileDB Core 2.3. Please reopen or ping us if you encounter any more issues.

**Comment by ihnorton** [comment]
Cross-reference initial fix for this in TileDB 2.2 which added normalization for reserved characters: https://github.com/TileDB-Inc/TileDB/pull/2047
